pubsub.go

```diff
@@ -232,6 +237,18 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 	return ps, nil
 }
 
+// WithPeerOutboundQueueSize is an option to set the buffer size for outbound messages to a peer
+// We start dropping messages to a peer if the outbound queue if full
+func WithPeerOutboundQueueSize(size int) Option {
+	return func(p *PubSub) error {
+		if size < 0 {
```

Comment:
It can't be zero either.

Reply:
You mean we never want it to be blocking ?

Reply:
@vyzo  Have addressed this. Let me know if you have any concerns before we merge this.